cnd/src/ethereum/mod.rs

```diff
@@ -16,3 +83,243 @@ impl IsStatusOk for TransactionReceipt {
+    pub gas_price: U256,
+    /// Gas amount
+    pub gas: U256,
+    /// Input data
+    pub input: Bytes,
+}
+
+/// A log produced by a transaction.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct Log {
+    /// H160
+    pub address: H160,
+    /// Topics
+    pub topics: Vec<H256>,
+    /// Data
+    pub data: Bytes,
+    /// Block Hash
+    #[serde(rename = "blockHash")]
+    pub block_hash: Option<H256>,
+    /// Block Number
+    #[serde(rename = "blockNumber")]
+    pub block_number: Option<U256>,
+    /// Transaction Hash
+    #[serde(rename = "transactionHash")]
+    pub transaction_hash: Option<H256>,
+    /// Transaction Index
+    #[serde(rename = "transactionIndex")]
+    pub transaction_index: Option<U256>,
+    /// Log Index in Block
+    #[serde(rename = "logIndex")]
+    pub log_index: Option<U256>,
+    /// Log Index in Transaction
+    #[serde(rename = "transactionLogIndex")]
+    pub transaction_log_index: Option<u64>,
+    /// Log Type
+    #[serde(rename = "logType")]
+    pub log_type: Option<String>,
+    /// Removed
+    pub removed: Option<bool>,
+}
+
+/// The block type returned from RPC calls.
+/// This is generic over a `TX` type.
+#[derive(Debug, Default, Clone, PartialEq, Deserialize, Serialize)]
+pub struct Block {
+    /// Hash of the block
+    pub hash: Option<H256>,
+    /// Hash of the parent
+    #[serde(rename = "parentHash")]
+    pub parent_hash: H256,
+    /// Hash of the uncles
+    #[serde(rename = "sha3Uncles")]
+    pub uncles_hash: H256,
+    /// Miner/author's address.
+    #[serde(rename = "miner")]
+    pub author: H160,
+    /// State root hash
+    #[serde(rename = "stateRoot")]
+    pub state_root: H256,
+    /// Transactions root hash
+    #[serde(rename = "transactionsRoot")]
+    pub transactions_root: H256,
+    /// Transactions receipts root hash
+    #[serde(rename = "receiptsRoot")]
+    pub receipts_root: H256,
+    /// Block number. None if pending.
+    pub number: Option<U128>,
+    /// Gas Used
+    #[serde(rename = "gasUsed")]
+    pub gas_used: U256,
+    /// Gas Limit
+    #[serde(rename = "gasLimit")]
+    pub gas_limit: U256,
+    /// Extra data
+    #[serde(rename = "extraData")]
+    pub extra_data: Bytes,
+    /// Logs bloom
+    #[serde(rename = "logsBloom")]
+    pub logs_bloom: H2048,
+    /// Timestamp
+    pub timestamp: U256,
+    /// Difficulty
+    pub difficulty: U256,
+    /// Total difficulty
+    #[serde(rename = "totalDifficulty")]
+    pub total_difficulty: U256,
+    /// Seal fields
+    #[serde(default, rename = "sealFields")]
+    pub seal_fields: Vec<Bytes>,
+    /// Uncles' hashes
+    pub uncles: Vec<H256>,
+    /// Transactions
+    pub transactions: Vec<Transaction>,
+    /// Size in bytes
+    pub size: Option<U256>,
+    /// Mix Hash
+    #[serde(rename = "mixHash")]
+    pub mix_hash: Option<H256>,
+    /// Nonce
+    pub nonce: Option<H64>,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum BlockNumber {
```

Comment:
Can we simplify this API based on our usage? I think we only ever use `Latest` here.